include/lbann/layers/activations/l2_loss.hpp

```diff
@@ -0,0 +1,58 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2014-2016, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory.
+// Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+// the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+//
+// LLNL-CODE-697807.
+// All rights reserved.
+//
+// This file is part of LBANN: Livermore Big Artificial Neural Network
+// Toolkit. For details, see http://software.llnl.gov/LBANN or
+// https://github.com/LLNL/LBANN.
+//
+// Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the license.
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef L2_NORM_HPP_INCLUDED
+#define L2_NORM_HPP_INCLUDED
+
+#include "lbann/layers/activations/activation.hpp"
+
+namespace lbann {
+
+/** Half the L2 loss of a tensor without the sqrt.
+ * (x^2)/2
+ * @todo relocate to loss layer as part of github issue 154
+ */
+template <data_layout T_layout>
+class l2_loss_layer : public entrywise_activation_layer {
```

Comment:
This is a special case of the `power_layer` added in #305. I think a dedicated L2 loss layer would be useful, but it should take both the prediction and the ground truth as inputs.

Reply:
The implementation mimics TF (https://www.tensorflow.org/api_docs/python/tf/nn/l2_loss), but yes, we can refactor to take both prediction and ground truth. We could also add scaling factor to power_layer for general purposes that would be useful for this.